main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"go_service/src"
+	"go_service/src/adapters"
+	"go_service/src/api"
+	"go_service/src/common"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+	"net/http"
+	"os"
+)
+
+func setupHandlers(cr common.CurrencyReader, sa api.SubscriberGateway) *http.ServeMux {
+	sm := http.NewServeMux()
```

Comment:
This is just **my** suggestion: instead of using the standard go tools, I would use a framework like gin. As it helps to avoid a lot of boilerplate code and facilitates routing.

Reply:
For this simple task, I decided not to complicate the project by adding dependencies such as http frameworks and cron libraries (besides the ORM and the migration tool). But I will use them later if the task is updated